classes/Tools.php

```diff
@@ -734,42 +738,96 @@ public static function getCldr(Context $context = null, $language_code = null)
     /**
      * Return price with currency sign for a given product.
      *
-     * @param float $price Product price
-     * @param object|array $currency Current currency (object, id_currency, NULL => context currency)
+     * @deprecated Since 1.7.5.0. Please use Locale::formatPrice() instead
+     * @see PrestaShop\PrestaShop\Core\Localization\Locale
+     *
+     * @param float $price
+     *                     Product price
+     * @param object|array $currency
+     *                               Current currency (object, id_currency, NULL => context currency)
+     * @param bool $no_utf8
+     *                      Not used anymore
+     * @param Context|null $context
      *
-     * @return string Price correctly formated (sign, decimal separator...)
+     * @return string Price correctly formatted (sign, decimal separator...)
      *                if you modify this function, don't forget to modify the Javascript function formatCurrency (in tools.js)
+     *
+     * @throws LocalizationException
      */
     public static function displayPrice($price, $currency = null, $no_utf8 = false, Context $context = null)
     {
+        @trigger_error(
+            'Tools::displayPrice() is deprecated since version 1.7.5.0. '
+            . 'Use PrestaShop\PrestaShop\Core\Localization\Locale::formatPrice() instead.',
+            E_USER_DEPRECATED
+        );
+
         if (!is_numeric($price)) {
             return $price;
         }
-        if (!$context) {
-            $context = Context::getContext();
+
+        $context = $context ?: Context::getContext();
+        $currency = $currency ?: $context->currency;
+
+        if (is_int($currency)) {
+            $currency = Currency::getCurrencyInstance($currency);
         }
-        if ($currency === null) {
-            $currency = $context->currency;
-        } elseif (is_int($currency)) {
-            $currency = Currency::getCurrencyInstance((int) $currency);
+
+        $locale = $context->currentLocale;
+        if (null === $locale) {
+            $locale = static::getCurrencyLocale($context);
         }
+        $currencyCode = is_array($currency) ? $currency['iso_code'] : $currency->iso_code;
 
-        $cldr = self::getCldr($context);
+        return $locale->formatPrice($price, $currencyCode);
+    }
+
+    protected static function getCurrencyLocale(Context $context)
```

Comment:
Why not inject the related service directly? (self::SERVICE_LOCALE_REPOSITORY)

getCurrencyLocale(ServiceLocaleRepository, Locale $locale) fits better your needs imho